airflow/api_fastapi/core_api/security.py

```diff
@@ -82,6 +82,23 @@ def callback():
     return inner
 
 
+def requires_access_asset(method: ResourceMethod) -> Callable:
+    def inner(
+        asset_id: str | None = None,
+        user: Annotated[BaseUser | None, Depends(get_user)] = None,
+    ) -> None:
+        def callback():
+            return get_auth_manager().is_authorized_asset(
+                method=method, details=AssetDetails(id=asset_id), user=user
+            )
+
+        _requires_access(
+            is_authorized_callback=callback,
+        )
+
+    return inner
```

Comment:
Maybe something like:
```python
def requires_access_assets(method: ResourceMethod) -> Callable:
    def inner(
            user: Annotated[BaseUser | None, Depends(get_user)] = None,
    ) -> None:
        def callback():
            return get_auth_manager().is_authorized_asset(
                method=method, details=AssetDetails(id=None), user=user
            )

        _requires_access(
            is_authorized_callback=callback,
        )

    return inner
```

So we do not define an extra `asset_id` param for endpoint that do not take an `asset_id` param. (listing assets and all)